upstairs/src/lib.rs

```diff
@@ -2124,8 +2125,9 @@ impl Downstairs {
 
     fn validate_unencrypted_read_response(
         response: &mut ReadResponse,
-    ) -> Result<(), CrucibleError> {
+    ) -> Result<Option<u64>, CrucibleError> {
```

Comment:
There is a path when the blocks are not written yet where there is no hash.  I could return 0 in that case, and
as long as another block never hashes to zero, it would be okay, but it seems like this is the exact case
that `Option` was made for.

Reply:
This is true - ship it! :)

Reply:
thanks for the review!